swarm/api/client/client.go

```diff
@@ -175,7 +175,10 @@ func (c *Client) UploadDirectory(dir, defaultPath, manifest string, toEncrypt bo
 	} else if !stat.IsDir() {
 		return "", fmt.Errorf("not a directory: %s", dir)
 	}
-	return c.TarUpload(manifest, &DirectoryUploader{dir, defaultPath}, toEncrypt)
+	if _, err := os.Stat(filepath.Join(dir, defaultPath)); err != nil {
+		return "", fmt.Errorf("default path: %v", err)
```

Comment:
This should probably be more informative in the typical case of `os.IsNotExist(err)`, e.g. `the default path "index.html" was not found in the upload directory "/path/to/dir"`